File: Examples/CSharp/Working-Shapes/Working-with-Icons-and-Pictures/ExtractAllImagesFromPage.cs
using System.IO;
using Aspose.Diagram;
using System;
namespace Aspose.Diagram.Examples.CSharp.Diagrams
{
    public class ExtractAllImagesFromPage
    {
        public static void Run()
        {
            // ExStart:ExtractAllImagesFromPage
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Call a Diagram class constructor to load a VSD diagram
            Diagram diagram = new Diagram(dataDir + "ExtractAllImagesFromPage.vsd");

            // Enter page index i.e. 0 for first one
            foreach (Shape shape in diagram.Pages[0].Shapes)
            {
                // Filter shapes by type Foreign
                if (shape.Type == Aspose.Diagram.TypeValue.Foreign)
                {
                    using (System.IO.MemoryStream stream = new System.IO.MemoryStream(shape.ForeignData.Value))
                    {
                        // Load memory stream into bitmap object
                        System.Drawing.Bitmap bitmap = new System.Drawing.Bitmap(stream);

                        // Save bmp here
                        bitmap.Save(dataDir + "ExtractAllImages" + shape.ID + "_Out.bmp");
                    }
                }
            }
            // ExEnd:ExtractAllImagesFromPage
        }
    }
}

File: Examples/VisualBasic/Working-Shapes/Working-with-Icons-and-Pictures/ExtractAllImagesFromPage.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Diagram
Imports System

Namespace Diagrams
    Public Class ExtractAllImagesFromPage
        Public Shared Sub Run()
            ' ExStart:ExtractAllImagesFromPage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Shapes()

            ' Call the diagram constructor to load diagram from a VSD file
            Dim diagram As New Diagram(dataDir & "ExtractAllImagesFromPage.vsd")

            ' Enter page index i.e. 0 for first one
            For Each shape As Shape In diagram.Pages(0).Shapes
                ' Filter shapes by type Foreign
                If shape.Type = Aspose.Diagram.TypeValue.Foreign Then
                    Using stream As New System.IO.MemoryStream(shape.ForeignData.Value)
                        ' Load memory stream into bitmap object
                        Dim bitmap As New System.Drawing.Bitmap(stream)

                        ' Save bmp here
                        bitmap.Save(dataDir & "ExtractAllImages" & shape.ID & ".bmp")
                    End Using
                End If
            Next shape
            ' ExEnd:ExtractAllImagesFromPage
        End Sub
    End Class
End Namespace